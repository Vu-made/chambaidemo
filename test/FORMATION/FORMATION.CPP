#include <bits/stdc++.h>
#define N 1000001
using namespace std;
int L[N], R[N];
int n, m;
void xuly1()
{
    int i,j;
    cin >> i >> j;
    L[i] = L[j];
    R[i] = j;
    R[L[j]] = i;
    L[j] = i;
}

void xuly2()
{
    int i,j;
    cin >> i >> j;
    L[i] = j;
    R[i] = R[j];
    L[R[j]] = i;
    R[j] = i;
}

void xuly3()
{
    int i;
    cin >> i;
    R[L[i]] = R[i];
    L[R[i]] = L[i];
}

void xuly4()
{
    int i;
    cin >> i;
    cout << L[i] << ' ' << R[i] << endl;
}
int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    freopen("FORMATION.INP","r",stdin);
    freopen("FORMATION.OUT","w",stdout);
    cin >> n >> m;
    for (int i=1; i<=m; i++)
    {
        int k;
        cin >> k;
        if (k == 1) xuly1();
        else if (k==2) xuly2();
        else if (k==3) xuly3();
        else xuly4();
    }
    return 0;
}
