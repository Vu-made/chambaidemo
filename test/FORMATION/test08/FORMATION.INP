416 131
1 235 1
3 235
1 73 1
1 308 73
3 308
2 297 1
1 17 297
1 124 1
3 17
1 100 124
4 124
3 297
4 1
2 378 124
2 387 124
2 369 124
1 206 1
3 387
2 397 378
2 38 100
3 206
2 42 100
3 1
4 73
4 397
2 92 42
2 238 42
1 364 238
4 42
2 257 238
1 415 92
3 364
1 10 397
1 237 38
2 115 100
2 366 237
2 250 257
3 415
4 124
3 92
4 397
1 219 237
4 219
3 100
3 397
1 157 238
2 34 124
3 157
3 115
2 384 10
2 94 238
1 165 237
1 43 250
4 38
2 332 94
4 250
1 310 219
4 42
3 237
3 219
1 54 378
2 413 54
4 124
4 369
3 332
3 378
4 384
1 146 310
1 319 38
3 54
4 366
4 124
3 238
2 330 384
2 6 43
2 170 369
3 146
2 320 38
4 384
2 158 320
2 207 250
1 5 42
2 198 413
2 31 34
3 170
1 269 320
2 126 250
1 251 165
3 330
4 158
3 126
2 236 251
2 277 6
3 207
2 238 277
2 283 34
1 13 124
3 124
3 366
3 269
1 233 277
4 38
3 320
2 128 310
1 81 413
3 94
2 349 369
3 384
3 369
1 32 310
4 5
1 17 283
2 74 10
4 81
3 42
1 379 198
3 257
1 58 283
3 73
4 74
4 5
3 43
2 125 250
2 152 32
2 112 277
2 42 238
4 128
4 198
1 340 310
3 250
4 319
