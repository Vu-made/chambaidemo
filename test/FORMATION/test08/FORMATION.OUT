100 1
124 0
0 100
378 0
100 364
38 369
10 0
250 237
366 124
43 219
73 238
38 34
34 54
10 0
165 319
38 34
10 330
320 124
319 320
73 42
349 413
10 0
0 43
310 251
379 10
165 38
